### PR TITLE
Additional checks when registering component

### PR DESCRIPTION
**Pull request by doerfli**

### CODE REVIEW

File: contracts/modules/RegistryController.sol [hunk 1]
```diff
@@ -188,9 +188,19 @@ contract RegistryController is
 
         require(
             EnumerableSet.length(_contractNames[_release]) < MAX_CONTRACTS,
-            "ERROR:REC-005:MAX_CONTRACTS_LIMIT"
+            "ERROR:REC-010:MAX_CONTRACTS_LIMIT"
         );
 
+        require(_contractsInRelease[_release] > 0, "ERROR:REC-011:RELEASE_UNKNOWN");
```
Comment (matthiaszimmermann) [inline contracts/modules/RegistryController.sol:194]: isn't a release initially empty?

File: tests/test_registry.py [hunk 2]
```diff
@@ -101,6 +101,28 @@ def test_registry_deregister(registry, owner):
...
+    with brownie.reverts("ERROR:REC-012:CONTRACT_NAME_EMPTY"):
+        registry.register("", tx1, {'from': owner})
+
+    registry.register(name1, tx1, {'from': owner})
+    with brownie.reverts("ERROR:REC-013:CONTRACT_NAME_EXISTS"):
+        registry.register(name1, tx2, {'from': owner})
+    
+    with brownie.reverts("ERROR:REC-014:CONTRACT_ADDRESS_ZERO"):
+        registry.register(name2, ZERO_ADDRESS, {'from': owner})
+
```
Comment (matthiaszimmermann) [inline tests/test_registry.py:123]: also add deregister of a non-existing module as edge case

**Review by matthiaszimmermann — CHANGES_REQUESTED** [review]
add deregister edge case to unit test

File: tests/test_registry.py (hunk 2 above)
Comment (doerfli) [reply tests/test_registry.py:123]: thats already there in another test (`test_registry_deregister`) https://github.com/etherisc/gif-contracts/blob/7843a6b3f9f9e0d4c8bb2b1a8f819f4aad8ad9fb/tests/test_registry.py#L91

File: contracts/modules/RegistryController.sol (hunk 1 above)
Comment (doerfli) [reply contracts/modules/RegistryController.sol:194]: oh yes it is. :-( Well then, we need some other way to validate the `_release` attribute.

Comment (doerfli) [reply contracts/modules/RegistryController.sol]: @matthiaszimmermann i pushed a workaround for the `prepareRelease` case. I don't like it, but i don't have a better idea right now. Can you please have a look, maybe you'll find something better.

**Review by matthiaszimmermann — APPROVED** [review]
